routes/cartlist.router.js

```diff
@@ -0,0 +1,43 @@
+const express = require("express");
+const router = express.Router();
+const Product = require("../model/products.model");
+
+router.get("/", (req, res) => {
+	Product.find({ cartListed: true })
+		.select("-__v")
+		.then((data) => res.json({ items: data }))
+		.catch((error) => console.error(error));
+});
+
+router.post("/:productId", async (req, res) => {
+	const { productId } = req.body;
+	const query = { _id: productId };
+	const foundProduct = await Product.findById(productId);
+	const update = {
+		$set: {
+			cartListed: !foundProduct.cartListed,
+			quantityAddedToCart: 1,
+		},
+	};
+	const options = { new: true };
```

Comment:
Place all the async DB operations in try catch block . so if any error occurs it sends response accordingly

Reply:
done